tests/test_templatetags.py

```diff
@@ -349,6 +349,39 @@ class Component(component.Component):
         with self.assertRaises(TemplateSyntaxError):
             template.render(Context({}))
 
+    def test_slot_name_fill_typo_gives_helpful_error_message(self):
+        component.registry.register(name="test1", component=SlottedComponent)
+
+        template = Template(
+            """
+            {% load component_tags %}
+            {% component_block "test1" %}
+                {% fill "haeder" %}
+                    Custom header
+                {% endfill %}
+                {% fill "main" %}
+                    main content
+                {% endfill %}
+            {% endcomponent_block %}
+        """
+        )
+        with self.assertRaises(TemplateSyntaxError):
+            try:
+                template.render(Context({}))
+            except TemplateSyntaxError as e:
+                self.assertEqual(
+                    (
+                        "Component 'test1' passed fill(s) referring to undefined slot(s).\n"
+                        "Bad fills: ['haeder'].\n"
+                        "\n"
+                        "Did you mean: \n"
+                        "- haeder: header?\n"
+                        "Other unfilled slots are: ['footer']."
+                    ),
```

Comment:
No big deal, but I'm thinking that this might look nicer with a triple quoted string instead? To handle the indentation I tend to use textwrap.dedent.

```suggestion
                    textwrap.dedent("""
                        Component 'test1' passed fill(s) referring to undefined slot(s).
                        Bad fills: ['haeder'].
                        
                        Did you mean:
                        - haeder: header?
                        Other unfilled slots are: ['footer'].
                    """),
```

Reply:
I didn't know this, nice!

Reply:
Note: You need an import too :)

Reply:
Right... 😆 Thanks!